google/release/ha_image_janitor.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import datetime
+import json
+import os
+import re
+import sys
+import yaml
+
+from google.cloud import storage
+from spinnaker.run import check_run_quick, run_quick
+
+
+RELEASED_VERSION_MATCHER = re.compile('^[0-9]+\.[0-9]+\.[0-9]+\.yml$')
+LATEST_UNVALIDATED_MATCHER = re.compile('.*-latest-unvalidated\.yml$')
+BUILT_VERSION_MATCHER = re.compile('.*-[0-9]+\-[0-9]+\-[0-9]+\-[0-9]+\.yml$')
+
+SERVICES = [
+  'clouddriver',
+  'deck',
+  'echo',
+  'front50',
+  'gate',
+  'igor',
+  'orca',
+  'rosco',
+  'fiat'
+]
+
+RELEASED_TAG_KEY = 'released'
```

Comment:
images are "published", not "released".
Could these tags have a value for the date they were published?
Eventually there could be a sunset or deprecated tag.
FWIW, GCE images use some standard tags which are used by pantheon for browsing so you dont see older images. We still want to keep them around but it might be worthwhile to use those tags too so that one could browse images in pantheon and see just the current.